src/email/interfaces/email.service.interface.ts

```diff
@@ -0,0 +1,5 @@
+import { Email } from "../dto/email.dto";
+
+export abstract class BaseEmailService {
```

Comment:
This shouldn't be named as interface. It should be email.service.ts and it should be under email folder, not email/interfaces.